ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/messages/TransactionsMessage.java

```diff
@@ -47,7 +49,24 @@ public static TransactionsMessage create(final Iterable<Transaction> transaction
     return new TransactionsMessage(tmp.encoded());
   }
 
-  private TransactionsMessage(final BytesValue data) {
+  public static LimitedTransactionsMessages createLimited(
+      final Iterable<Transaction> transactions, final TransactionsMessageLimiter limiter) {
+    final Set<Transaction> includedTransactions = new HashSet<>();
+    final BytesValueRLPOutput tmp = new BytesValueRLPOutput();
+    tmp.startList();
+    for (final Transaction transaction : transactions) {
+      transaction.writeTo(tmp);
+      includedTransactions.add(transaction);
+      if (limiter.isLastTransactionToIncludeWithoutExceedingLimit(tmp)) {
```

Comment:
Do we really need the extra complexity of this `Limiter` class when it just runs a >= check?  I think it would be simpler to just pass the size constraints to this method.

Reply:
I think yes. I mean in terms of being future proof if we want to have multiple policies, it might then be configurable through multiple instances of the interface. And for testing purpose it is easier to mock the Limiter interface.

Reply:
I don't see how this makes the code easier to test.  In terms of future-proofing, one of our design principles is YAGNI (You Ain't Gonna Need It) :).  I think it's better to start simple and make the code more complex as needed.

Reply:
Not a blocker though^ . I'll leave it up to you.

Reply:
I have simplified according to your idea of using a FunctionalInterface